edc-extensions/bpn-validation/bpn-validation-core/src/test/java/org/eclipse/tractusx/edc/validation/businesspartner/functions/BusinessPartnerGroupFunctionTest.java

```diff
@@ -122,24 +124,46 @@ void evaluate_validOperator(String ignored, Operator operator, List<String> assi
         assertThat(function.evaluate(operator, allowedGroups, createPermission(operator, allowedGroups), context)).isEqualTo(expectedOutcome);
     }
 
-    @Test
-    void evaluate_noEntryForBpn() {
-        var operator = NEQ;
+    @EnumSource(value = Operator.class, names = {"NEQ", "IS_NONE_OF"})
+    @ParameterizedTest
+    void evaluate_noEntryForBpnWithNegativeOperator_shouldBeTrue(Operator operator) {
+        var allowedGroups = List.of(TEST_GROUP_1, TEST_GROUP_2);
+        when(context.getContextData(eq(ParticipantAgent.class))).thenReturn(new ParticipantAgent(Map.of(), Map.of(PARTICIPANT_IDENTITY, TEST_BPN)));
+        when(store.resolveForBpn(TEST_BPN)).thenReturn(StoreResult.notFound("foobar"));
+
+        assertThat(function.evaluate(operator, allowedGroups, createPermission(operator, allowedGroups), context)).isTrue();
+    }
+
+    @EnumSource(value = Operator.class, names = {"NEQ", "IS_NONE_OF"})
```

Comment:
Test coverage is insufficient. you're only testing one scenario, i.e. left-hand operand is empty, right-hand operand is non-empty.
What about `[a,b,c] NEQ []`, `[a,b,c] IS_NONE_OF []`? same goes for the "positive" operator test.

Reply:
I added the test scenarios.